### PR TITLE
Jackson-based JSON encoding and decoding (WIP)

### PR DESCRIPTION
**Pull request by miikka**
This may or may not be a bit faster than Cheshire. At least it's not too much code.

* How to deal with custom types and other configuration?
* Needs tests.
* Needs proper benchmarking.

The current `muuntaja.json` should be probably thrown away and this should replace it.

### CODE REVIEW

**Comment by ikitommi** [comment]
:+1:

File: test/muuntaja/jackson_test.clj [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                  :timestamp "1970-01-01T00:00:00Z"}]
+
+    (testing "cheshire"
+      (is (= expected (cheshire/parse-string
+                        (cheshire/generate-string data)
+                        true))))
+
+    #_(testing "jackson"
+        (is (= (cheshire/generate-string data)
+               (jackson/to-json data)))
```
Comment (miikka) [inline test/muuntaja/jackson_test.clj]: This is a bit tricky assertion in that the same JSON object has multiple string representations. For example, `{"a":1}` is equivalent to `{"a": 1}`, `{"a": 1, "b": 2}` is equivalent to `{"b": 2, "a": 1}` and `1` is equivalent to `1.0`. If it does not work, it's probably not worthwhile to spend a lot of effort on it.

Alternatively we could [canonicalize our JSON like this](http://hypothesis.works/articles/canonical-serialization/):

```clojure
(defn make-canonical [s] (-> s jackson/from-json jackson/to-json))
(deftest test-canonical-is-canonical
  (is (= (make-canonical (make-canonical some-json)) (make-canonical some-json))))
```

Comment (ikitommi) [inline test/muuntaja/jackson_test.clj]: Good point. Comparing JSON strings is a bad idea. I had no plans to do more here, just committed my learning tests.

**Comment by miikka** [comment]
@ikitommi, this PR is getting quite large already and I'd like to merge it soon. Do you think it's okay to merge? I made your tests work and added basic support for keywordizing the map keys and for custom encoders.

**Comment by ikitommi** [comment]
Go ahead, looks great!

**Pull request merged**